pkgs/development/compilers/chez-racket/shared.nix

```diff
@@ -0,0 +1,41 @@
+args:
+{ stdenv, lib, fetchFromGitHub, coreutils, darwin
+, ncurses, libiconv, libX11, zlib, lz4
+}:
+
+stdenv.mkDerivation (args // {
+  version = "unstable-2021-12-11";
+
+  src = fetchFromGitHub {
+    owner  = "racket";
+    repo   = "ChezScheme";
+    rev    = "8846c96b08561f05a937d5ecfe4edc96cc99be39";
+    sha256 = "IYJQzT88T8kFahx2BusDOyzz6lQDCbZIfSz9rZoNF7A=";
+    fetchSubmodules = true;
+  };
+
+  prePatch = ''
+    rm -rf zlib/*.c lz4/lib/*.c
+  '';
+
+  postPatch = ''
+    export ZLIB="$(find ${zlib.out}/lib -type f | sort | head -n1)"
+    export LZ4="$(find ${lz4.out}/lib -type f | sort | head -n1)"
+  '';
+
+  nativeBuildInputs = lib.optional stdenv.isDarwin darwin.cctools;
```

Comment:
I was able to get this working on aarch64-darwin by changing this to
```nix
nativeBuildInputs = lib.optional stdenv.isDarwin [darwin.cctools darwin.autoSignDarwinBinariesHook];
```

Ran the REPL and voila:
```
[nix-shell:~/.cache/nixpkgs-review/rev-a9f68fe379318d19830db01846db215359bb8598]$ ./results/chez-racket/bin/scheme
Chez Scheme Version 9.5.7.2
Copyright 1984-2021 Cisco Systems, Inc.

> (+ 3 4)
7
> (exit)
```

Reply:
I got the hint for this fix from https://github.com/NixOS/nixpkgs/issues/149692#issuecomment-989086581

Reply:
Thanks! I'll commit this.